### ISSUE TITLE
Guidance for when it is appropriate to edit or delete a GitHub comment

### ISSUE THREAD

**Issue by nigelmegitt** [w3c/PWETF#23]
GitHub is increasingly being used for W3C work, and has a feature that comments in issues or pull requests can be edited or deleted, not only by the person who wrote them but by others also. There doesn't seem to be much control available about who can do what; it is quite "open" by default at the moment.

A couple of times I have observed a WG member editing or deleting comments made both by that member and by other members. There seemed to be a strong reaction from other members of the WG that this was not appropriate behaviour, but the person doing it noted that there is no policy applying to this, nor is there any established norm, and furthermore, explained their reasons for doing so.

I would very strongly like to have some guidance as part of the CEPC about the treatment of comments made by others in GitHub (or any other forum) to clarify this. This seems to be a matter of professional respect primarily, but also it can get more serious if an allegation of a breach of the CEPC is made via comments which are then edited or removed after the fact. Such editing/removal needs itself to be a breach of the CEPC.

Needless to say, I'd welcome other thoughts on this.

**Comment by dauwhe** [comment]
I have edited other people's GitHub comments a few times, but only to fix structural issues. Most commonly, people type html tags but don't enclose them in code blocks, and they disappear, making the comment hard to understand. Or the quoting gets messed up…

I would not be pleased if someone made substantive edits to my own GitHub comments.

**Comment by nigelmegitt** [comment]
Indeed, same here, and I _think_ that's the general view, but it appears that it is not universal. The idea of deleting other people's comments without prior agreement feels _very wrong_ to me but it is hard to articulate exactly why. I'd be interested to know the range (and frequency) of views on this.

**Comment by vlevantovsky** [comment]
I agree with both points but, frankly, I am not sure why this is an issue - the GitHub doesn't allow me [a regular participant in a discussion] to edit someone else's comments, and I hope the same is true for others. If someone is given a superpower to edit the whole thread, that power should come with the responsibility to be mindful and sensitive, and we'd probably know who has that power.

**Comment by nigelmegitt** [comment]
@vlevantovsky In fact if you have write access to a repo then you can edit or delete comments, unfortunately.

**Comment by nigelmegitt** [comment]
@vlevantovsky you'll see I just edited your [comment](https://github.com/w3c/PWETF/issues/23#issuecomment-453161361) above. I could have deleted it, and I don't think there's a way to prevent this unless you don't ever grant write access to a repo, which would make the system a lot harder to use.

**Comment by vlevantovsky** [comment]
Ok, I see your point (and I did believe you when you replied to my comment, without "power demo"). :-)

**Comment by vlevantovsky** [comment]
I don't have a strong opinion on this issue, but I think it is next to impossible to try and regulate every aspect of our lives. I am not sure CEPC is the place to do it either. For as long as W3C participants adhere to the CEPC, the issue of "bad comment" may never come up, and if and when it does, I trust that group chairs would have sufficient privileges and the right judgement call to deal with it.

**Comment by nigelmegitt** [comment]
> chairs would have sufficient privileges

In fact the issue is not about if chairs have _enough_ privileges but the right granularity of privileges. It is possible to remove write access to repos altogether, but this is a sledgehammer all-or-nothing solution. It is currently not possible to remove comment editing rights on their own while still allowing other contributions.

Also, after the fact it is next to impossible to deal with it. Say for example that I as Chair judge that a comment should be reinstated. If it has been deleted, this is impossible.

My reason for raising this issue was because the response I received from a group member who did this was "it doesn't say I can't anywhere, and GitHub allows me to, so it must be okay". _If_ we have consensus here that deleting other people's comments should not be the norm, then unless we write that down somewhere formally, this behaviour is likely to continue and be impossible to challenge.

That said, it is a big 'if' and I am not sure if we do indeed have that consensus. Hence the request for views.
